thirdparty/faiss/faiss/IndexCosine.cpp

```diff
@@ -447,20 +447,24 @@ DistanceComputer* IndexProductResidualQuantizerCosine::get_distance_computer() c
 //
 IndexHNSWFlatCosine::IndexHNSWFlatCosine() {
     is_trained = true;
+    is_cosine = true;
 }
 
 IndexHNSWFlatCosine::IndexHNSWFlatCosine(int d, int M) :
     IndexHNSW(new IndexFlatCosine(d), M) 
 {
     own_fields = true;
     is_trained = true;
+    is_cosine = true;
```

Comment:
Can I ask where does this get used?

Reply:
this was added for a consistency: if a storage IndexHNSW is cosine-based, then IndexHNSW is also cosine-based